<?xml version="1.0" encoding="utf-8"?>
<launch>
    <!-- Required arguments -->
    <arg name="controllers_ns"      default="anymal_d"/>
    <arg name="controllers_args"    default="joint_controller"/>
    <arg name="tbai_config_path"    default="$(find tbai_dtc)/config/default_config.yaml"/>
    <arg name="description_name"    default="robot_description"/>
    <arg name="description_file"    default="$(find tbai_description)/urdf/anymal_d_gazebo.urdf.xacro"/>

    <!-- Gazebo parameters -->
    <arg name="gui" default="false"/>
    <arg name="paused" default="false"/>
    <arg name="world" default="normal"/>
    <arg name="x" default="0.0"/>
    <arg name="y" default="0.0"/>
    <arg name="z" default="3.0"/>

    <!-- Upload config path to ROS param server -->
    <param name="tbai_config_path" type="string" value="$(arg tbai_config_path)"/>

      <arg name="joystick_dev" default="/dev/input/js0"/>
    <node name="joy_node" pkg="joy" type="joy_node" output="screen">
        <param name="dev" value="$(arg joystick_dev)"/>
    </node>

    <!-- Launch gazebo -->
    <include file="$(find tbai_gazebo)/launch/world.launch">
        <arg name="controllers_ns" value="$(arg controllers_ns)"/>
        <arg name="controllers_args" value="$(arg controllers_args)"/>
        <arg name="config_file" value="$(arg tbai_config_path)"/>
        <arg name="description_name" value="$(arg description_name)"/>
        <arg name="description_file" value="$(arg description_file)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="world" value="$(arg world)"/>
        <arg name="x" value="$(arg x)"/>
        <arg name="y" value="$(arg y)"/>
        <arg name="z" value="$(arg z)"/>
        <arg name="paused" value="$(arg paused)"/>
    </include>

    <!-- Launch simple node -->
    <node name="simple" pkg="tbai_dtc" type="simple" output="screen"/>

    <param name="taskFile"          value="$(find tbai_dtc)/config/task.info" />
    <param name="referenceFile"     value="$(find tbai_dtc)/config/reference.info" />
    <param name="gaitCommandFile"   value="$(find tbai_dtc)/config/gait.info"/>
    <param name="dtcModelFile"      value="$(find tbai_dtc)/models/dtc_wbc_jitted_wbc07.pt"/>
    <param name="urdfFile"          value="$(find ocs2_robotic_assets)/resources/anymal_d/urdf/anymal.urdf" />

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_sqp_mpc" name="legged_robot_sqp_mpc"
          output="screen" launch-prefix=""/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
            output="screen" launch-prefix="gnome-terminal --"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_target" name="legged_robot_target"
          output="screen" launch-prefix="gnome-terminal --"/>

    <arg name="rvizconfig" default="$(find tbai_dtc)/launch/rviz/dtc.rviz" />
      <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)" output="screen" />   
</launch>